experimental/piranha_playground/rule_inference/static_inference.py

```diff
@@ -18,7 +19,64 @@
 from piranha_playground.rule_inference.utils.node_utils import NodeUtils
 from piranha_playground.rule_inference.utils.rule_utils import RawRule
 
-from piranha_playground.rule_inference.template_parser import WILDCARD
+
+@attr.s
+class ChildProcessingStrategy(ABC):
+    """
+    Abstract Base Class to define a strategy to process a named child node.
+    """
+
+    query_writer = attr.ib(type="QueryWriter")
+
+    @abstractmethod
+    def process(self, cursor: TreeCursor, depth: int) -> str:
+        pass
+
+
+@attr.s
+class SimplifyStrategy(ChildProcessingStrategy):
+    """
+    Strategy to simplify the processing of a named child node.
```

Comment:
what does simplify imply here? 

This strategy is related to adding alternations right?


aah I see this is that smart simplification ?

Reply:
This is for when you want simplify queries (match only current node and assign groups to its children). It’s a refactoring because the code was messy

Reply:
I see. 
Let's document this method a bit